### PR TITLE
MITK upgrade

### PR DESCRIPTION
**Pull request by JostMigenda**
This combines work from the previous MITK upgrade effort and @alonsoJASL’s fork … and then rebased all of it onto of the current `development` branch. This was a bit of a painful rebase—going forward, we really should avoid having branches diverge for multiple years!—but José and I completed it this afternoon.

Next up, I’ll try to build it on my machine and see what further changes are needed. Therefore, I’ll leave this as a draft PR for now.

### CODE REVIEW

File: CemrgApp/Applications/MainApp/target_libraries.cmake
Comment (OrodRazeghi) [inline CemrgApp/Applications/MainApp/target_libraries.cmake:10]: What was the difference between dicom and dicombrowser?

File: CemrgApp/CMakeExternals/VMTK.cmake
Comment (OrodRazeghi) [inline CemrgApp/CMakeExternals/VMTK.cmake]: We should document somewhere why we are temporarily using a forked `vmtk` and not the official releases.

File: CemrgApp/Plugins/kcl.cemrgapp.easi/src/internal/EASIView.cpp
Comment (OrodRazeghi) [inline CemrgApp/Plugins/kcl.cemrgapp.easi/src/internal/EASIView.cpp:580]: This plugin is not in use, right? It has hardcoded paths to my old home directory. It should probably be checked.

File: CemrgApp/Plugins/kcl.cemrgapp.scar/src/internal/YZSegView.cpp
Comment (OrodRazeghi) [inline CemrgApp/Plugins/kcl.cemrgapp.scar/src/internal/YZSegView.cpp:117]: Editor was replaced by browser?

File: CemrgApp/Applications/MainApp/target_libraries.cmake
Comment (JostMigenda) [reply CemrgApp/Applications/MainApp/target_libraries.cmake:10]: It was renamed in v2021.02, see https://phabricator.mitk.org/T27001

**Review by alonsoJASL — DISMISSED** [review]
Changes look OK.

File: CemrgApp/Plugins/kcl.cemrgapp.scar/src/internal/YZSegView.cpp
Comment (alonsoJASL) [reply CemrgApp/Plugins/kcl.cemrgapp.scar/src/internal/YZSegView.cpp:117]: This is a relatively old change MITK did. One needs to call `dicombrowser` instead of `dicomeditor`

File: CemrgApp/Plugins/kcl.cemrgapp.easi/src/internal/EASIView.cpp
Comment (alonsoJASL) [reply CemrgApp/Plugins/kcl.cemrgapp.easi/src/internal/EASIView.cpp:580]: This plugin is not used. It could be that we could just get rid of it or add something like a user-defined path. In my opinion, since it's not being used at the moment, we could leave that for another PR.

File: CemrgApp/CMakeExternals/VMTK.cmake
Comment (alonsoJASL) [reply CemrgApp/CMakeExternals/VMTK.cmake]: Do you think in the wiki would suffice or in a comment on this CMake file?

File: .github/workflows/build.yml
Comment (LouiseABowler) [inline .github/workflows/build.yml]: Looks like this section was duplicated during the rebase, lines 8-20 are identical

File: .github/workflows/test.yml
Comment (LouiseABowler) [inline .github/workflows/test.yml]: Another duplicated section from the rebase here

File: CemrgApp/CMakeExternals/VMTK.cmake
Comment (LouiseABowler) [reply CemrgApp/CMakeExternals/VMTK.cmake]: The wiki would be the best place for that, I think. It's certainly easier to find than knowing to look exactly here, deep in the cmake files!

Comment (LouiseABowler) [inline CemrgApp/CMakeExternals/VMTK.cmake]:
```suggestion
    set(VMTK_GIT_REPOSITORY "https://github.com/OpenHeartDevelopers/vmtk.git" CACHE STRING "The git repository for cloning VMTK")
```

`CemrgDevelopers` still works but redirects to `OpenHeartDevelopers`. The org must have been renamed at some point, so let's use the most recent name

File: CemrgApp/Plugins/kcl.cemrgapp.mmcwplugin/src/internal/MmcwView.cpp
Comment (LouiseABowler) [inline CemrgApp/Plugins/kcl.cemrgapp.mmcwplugin/src/internal/MmcwView.cpp:686]: On a similar line further up, `QDialog* inputs` was changed to `QDialog *inputs` - want to do the same here for consistency?

File: CemrgApp/Plugins/kcl.cemrgapp.easi/src/internal/EASIView.cpp
Comment (LouiseABowler) [reply CemrgApp/Plugins/kcl.cemrgapp.easi/src/internal/EASIView.cpp:580]: Is there a GitHub issue for this? I also spotted a few hardcoded paths in another file in this PR, it isn't within scope to fix now but we should at some point.

File: CemrgApp/Modules/CemrgAppModule/src/CemrgTests.cpp
Comment (LouiseABowler) [inline CemrgApp/Modules/CemrgAppModule/src/CemrgTests.cpp:88]: Not one to fix now, but could you make an issue to remove these hardcoded paths?

**Review by LouiseABowler — DISMISSED** [review]
This is looking really good @JostMigenda! 🚀 

I've left a couple of comments but go ahead and merge this when you're ready, those comments won't need any further review.

I have also given this PR a go on my Linux machine today. Mixed results there - it builds fine, but the applcation won't open. I suspect any fixes will inolve the setup instructions rather than the code, do don't let that block this PR 👍

File: CemrgApp/Plugins/kcl.cemrgapp.easi/src/internal/EASIView.cpp
Comment (JostMigenda) [reply CemrgApp/Plugins/kcl.cemrgapp.easi/src/internal/EASIView.cpp:580]: I’ve created #81 to track this.

File: CemrgApp/Modules/CemrgAppModule/src/CemrgTests.cpp
Comment (JostMigenda) [reply CemrgApp/Modules/CemrgAppModule/src/CemrgTests.cpp:88]: Added this to #81.

File: CemrgApp/Plugins/kcl.cemrgapp.mmcwplugin/src/internal/MmcwView.cpp
Comment (JostMigenda) [reply CemrgApp/Plugins/kcl.cemrgapp.mmcwplugin/src/internal/MmcwView.cpp:686]: I have no strong feelings either way and would just leave this as-is. (It’s probably inconsistent in a bunch of other places across the codebase as well; so if we do want to make it consistent, let’s use something like [clang-format](https://clang.llvm.org/docs/ClangFormat.html) instead of manual edits.)

**Review by LouiseABowler — APPROVED** [review]
Thanks for the revisions @JostMigenda, this looks good to me 🙌

File: CemrgApp/CMakeExternals/VMTK.cmake
Comment (JostMigenda) [reply CemrgApp/CMakeExternals/VMTK.cmake]: I couldn’t find a good page on the wiki where this fit, so I posted it as an issue in #82 instead. That’s also a good place if we need to discuss more.

**Comment by JostMigenda** [comment]
Test’s are failing during the “Download precompiled Build folder” step; no logs for that step is available, but the error message for the overall workflow run says:
`System.IO.IOException: No space left on device`
Is it possible that the downloaded build folder (plus the CemrgApp repo itself) exceeds the [14 GB disk space limit](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories) on GitHub’s runners?

Anyway, since the tests already fail on the head of the `development` branch, this shouldn’t stop us from merging this PR.

**Pull request merged**
